float32 fromx
float32 fromy
float32 tox
float32 toy
float32 size